apisix/plugins/cors.lua

```diff
@@ -179,8 +193,49 @@ function _M.header_filter(conf, ctx)
         end
     end
 
-    ctx.cors_allow_origins = allow_origins
-    set_cors_headers(conf, ctx)
+    return allow_origins
+end
+
+local function process_with_allow_origins_by_regex(conf, ctx)
+    if conf.allow_origins_by_regex == nil  or next(conf.allow_origins_by_regex) == nil then
+        return
+    end
+    local req_origin = core.request.header(ctx, "Origin")
+
+    if not conf.allow_origins_by_regex_rules_concat then
+        local allow_origins_by_regex_rules = {}
```

Comment:
We can create the table with pre-allocated array part size to `#conf.allow_origins_by_regex `.